python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -474,16 +473,26 @@ def eq(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         else:
             from pyspark.pandas.base import column_op
 
-            Column = ConnectColumn if is_remote() else PySparkColumn
-            return column_op(cast(Callable[..., GenericColumn], Column.__eq__))(left, right)
+            if is_remote():
+                from pyspark.sql.connect.column import Column as ConnectColumn
+
+                Column = ConnectColumn
+            else:
+                Column = PySparkColumn  # type: ignore[assignment]
+            return column_op(Column.__eq__)(left, right)  # type: ignore[arg-type]
```

Comment:
We should remove this away also by using `lambda` instead. Let's fix them all in another PR.

Reply:
Sure. Just created ticket for address this one: SPARK-43159